tests/BatchedProcessorTest.php

```diff
@@ -100,7 +138,60 @@ public function testBatching() {
 		$this->assertEquals(9, $data->currentStep);
 		$this->assertEquals(42, count($index->getAdded()));
 		$this->assertTrue($data->isComplete);
+
+		// Check any additional queued jobs
+		$processor->afterComplete();
+		$service = singleton('QueuedJobService');
+		$jobs = $service->getJobs();
+		$this->assertEquals(1, count($jobs));
+		$this->assertInstanceOf('SearchUpdateCommitJobProcessor', $jobs[0]['job']);
+	}
+
+	/**
+	 * Test creation of multiple commit jobs
+	 */
+	public function testMultipleCommits() {
+		$index = singleton('BatchedProcessorTest_Index');
+		$index->reset();
+
+		// Test that running a commit immediately after submitting to the indexes
+		// correctly commits
+		$first = SearchUpdateCommitJobProcessor::queue();
+		$second = SearchUpdateCommitJobProcessor::queue();
+
+		$this->assertFalse($index->getIsCommitted());
+
+		// First process will cause the commit
+		$this->assertFalse($first->jobFinished());
+		$first->process();
+		$allMessages = $first->getMessages();
+		$this->assertTrue($index->getIsCommitted());
+		$this->assertTrue($first->jobFinished());
+		$this->assertStringEndsWith('All indexes committed', $allMessages[2]);
+
+		// Executing the subsequent processor should not re-trigger a commit
+		$index->reset();
+		$this->assertFalse($second->jobFinished());
+		$second->process();
+		$allMessages = $second->getMessages();
+		$this->assertFalse($index->getIsCommitted());
```

Comment:
@patbolo I only temporarily reset the $index to set comitted to false to prove that it's not re-comitted by `$second`. In practice it would stay comitted, although just for the sake of this test we toggle it back to false.